src/main/java/net/dv8tion/jda/api/interactions/components/text/TextInput.java

```diff
@@ -0,0 +1,481 @@
+         */
+        public Builder setLabel(@Nonnull String label)
+        {
+            Checks.notBlank(label, "Label");
+            Checks.notLonger(label, LABEL_MAX_LENGTH, "Label");
+            this.label = label;
+            return this;
+        }
+
+        /**
+         * Sets the style for this TextInput
+         * <br>Possible values are:
+         * <ul>
+         *     <li>{@link TextInputStyle#SHORT}</li>
+         *     <li>{@link TextInputStyle#PARAGRAPH}</li>
+         * </ul>
+         *
+         * @param  style
+         *         The style to set
+         *
+         * @throws IllegalArgumentException
+         *         If style is null or UNKNOWN
+         *
+         * @return The same Builder for chaining convenience.
+         */
+        public Builder setStyle(TextInputStyle style)
+        {
+            Checks.notNull(style, "Style");
+            Checks.check(style != TextInputStyle.UNKNOWN, "TextInputStyle cannot be UNKNOWN!");
+            this.style = style;
+            return this;
+        }
+
+        /**
+         * Sets whether the user is required to write in this TextInput.
+         *
+         * @param  required 
+         *         If this TextInput should be required
+         *
+         * @return The same builder instance for chaining
+         */
+        @Nonnull
+        public Builder setRequired(boolean required)
+        {
+            this.required = required;
+            return this;
+        }
+
+        /**
+         * Sets the minimum length of this input field.
+         *
+         * <b>This has to be between 0 and {@value #TEXT_INPUT_MAX_LENGTH}</b>
+         *
+         * @param  minLength
+         *         The minimum amount of characters that need to be written
+         *
+         * @throws IllegalArgumentException
+         *         If minLength is negative or greater than {@value #TEXT_INPUT_MAX_LENGTH}
+         *
+         * @return The same builder instance for chaining
+         */
+        @Nonnull
+        public Builder setMinLength(int minLength)
+        {
+            Checks.notNegative(minLength, "Minimum length");
+            Checks.check(minLength <= TEXT_INPUT_MAX_LENGTH, "Minimum length cannot be longer than %d characters!", TEXT_INPUT_MAX_LENGTH);
+            this.minLength = minLength;
+            return this;
+        }
+
+        /**
+         * Sets the maximum length of this input field.
+         *
+         * <b>This has to be between {@value #TEXT_INPUT_MIN_LENGTH} and {@value #TEXT_INPUT_MAX_LENGTH}</b>
+         *
+         * @param maxLength The maximum amount of characters that need to be written
+         *
+         * @throws IllegalArgumentException
+         *         If maxLength is smaller than {@value #TEXT_INPUT_MIN_LENGTH} or greater than {@value #TEXT_INPUT_MAX_LENGTH}
+         *
+         * @return The same builder instance for chaining
+         */
+        @Nonnull
+        public Builder setMaxLength(int maxLength)
```

Comment:
You should specify what the defaults for all these setters are, since its not obvious for the user.